wallet/src/main/scala/org/bitcoins/wallet/internal/AddressHandling.scala

```diff
@@ -410,7 +410,8 @@ private[wallet] trait AddressHandling extends WalletLogger {
       address: BitcoinAddress,
       tag: AddressTag): Future[AddressTagDb] = {
     val addressTagDb = AddressTagDb(address, tag)
-    addressTagDAO.upsert(addressTagDb)
+    val f = addressTagDAO.create(addressTagDb)
+    f
```

Comment:
From my understanding, this is what we are trying to get away from akin to #1823, with this `val` we may start the future before we want to

Reply:
Let's leave this aside for now and continue this on #1823 , is there a reason we were using `upsert()` rather than `create`?

Reply:
The reason I did that originally was in case someone tried to tag the address with the same tag twice, then they wouldn't get duplicates

Reply:
I think this should be fine though

Reply:
Ok, but then why not enforce with this with a primary key at the database level of (BitcoinAddress,Tag)? This unit test does not pass otherwise because with the use of `upsert()` an address can have at most one tag

Reply:
Yeah, that's probably best. I forget why I didn't do that originally but that makes the most sense to me